test/auth/auth.e2e-spec.ts

```diff
@@ -66,4 +67,45 @@ describe("AuthController (e2e)", () => {
         .expect(HttpStatus.UNAUTHORIZED);
     });
   });
+
+  describe("POST /auth/register/student", () => {
+    it("registers a student and returns CREATED(201) with auth token", async () => {
+      const studentData = createStudentRegistrationData();
```

Comment:
Since we are reusing this in every `it` block, let's define this inside a `beforeEach` inside the `describe` block for this endpoint.